core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -128,9 +128,22 @@ object StorageTool extends Logging {
       setIgnoreFormatted(namespace.getBoolean("ignore_formatted")).
       setControllerListenerName(config.controllerListenerNames.get(0)).
       setMetadataLogDirectory(config.metadataLogDir)
-    Option(namespace.getString("release_version")).foreach(
-      releaseVersion => formatter.
-        setReleaseVersion(MetadataVersion.fromVersionString(releaseVersion)))
+
+    def metadataVersionsToString(first: MetadataVersion, last: MetadataVersion): String = {
+      val versions = MetadataVersion.VERSIONS.slice(first.ordinal, last.ordinal + 1)
+      versions.map(_.toString).mkString(", ")
+    }
+    Option(namespace.getString("release_version")).foreach(releaseVersion => {
+      try {
+        formatter.setReleaseVersion(MetadataVersion.fromVersionString(releaseVersion))
+      } catch {
+        case _: Throwable =>
+          throw new TerseFailure("Unknown metadata.version " + releaseVersion +
+          ". Supported metadata.version are " + metadataVersionsToString(
+            MetadataVersion.MINIMUM_VERSION, MetadataVersion.latestProduction()))
+      }
```

Comment:
Could we use an interpolated string here?
```suggestion
          throw new TerseFailure(s"Unknown metadata.version $releaseVersion. Supported metadata.version are ${
            metadataVersionsToString(
              MetadataVersion.MINIMUM_VERSION, MetadataVersion.latestProduction())
          }")
```

Reply:
good catch